src/Domain/TokenGenerator.php

```diff
@@ -0,0 +1,26 @@
+<?php
+
+declare( strict_types = 1 );
+
+namespace WMDE\Fundraising\Frontend\Domain;
+
+/**
+ * @licence GNU GPL v2+
+ * @author Kai Nissen < kai.nissen@wikimedia.de >
+ */
+class TokenGenerator {
+
+	public function __construct( int $tokenLength, string $validityTimeSpan ) {
+		$this->tokenLength = $tokenLength;
+		$this->validityTimeSpan = $validityTimeSpan;
+	}
+
+	public function generateToken() {
+		return bin2hex( random_bytes( $this->tokenLength ) );
+	}
+
+	public function generateTokenExpiry() {
```

Comment:
I think this should rather generate a `DateTime` object, for easier comparison later. The serialization could then be done in the Repository.
On the other hand, the check if the token is expired has to be done with a serialized date coming the from the data, so maybe it's better for consistency to always use a serialized date.
What's your opinion, @JeroenDeDauw ?